File: src/ocaml-output/FStar_Tests_Test.ml
open Prims
let main : 'uuuuu 'uuuuu1 . 'uuuuu -> 'uuuuu1 =
  fun argv ->
    FStar_Compiler_Util.print_string "Initializing ...\n";
    (try
       (fun uu___1 ->
          match () with
          | () ->
              (FStar_Main.setup_hooks ();
               (let uu___4 = FStar_Tests_Pars.init () in
                FStar_Compiler_Effect.op_Bar_Greater uu___4
                  (fun uu___5 -> ()));
               FStar_Tests_Norm.run_all ();
               (let uu___6 = FStar_Tests_Unif.run_all () in
                if uu___6
                then ()
                else FStar_Compiler_Effect.exit Prims.int_one);
               FStar_Compiler_Effect.exit Prims.int_zero)) ()
     with
     | FStar_Errors.Error (err, msg, r, _ctx) when
         let uu___2 = FStar_Options.trace_error () in
         FStar_Compiler_Effect.op_Less_Bar Prims.op_Negation uu___2 ->
         (if r = FStar_Compiler_Range.dummyRange
          then FStar_Compiler_Util.print_string msg
          else
            (let uu___4 = FStar_Compiler_Range.string_of_range r in
             FStar_Compiler_Util.print2 "%s: %s\n" uu___4 msg);
          FStar_Compiler_Effect.exit Prims.int_one))

File: src/tests/boot/FStar.Tests.Test.fs
#light "off"
module FStar.Tests.Test
open FStar.Compiler.Effect
open FStar.Syntax
open FStar.Errors
module S = FStar.Syntax.Syntax
module SS = FStar.Syntax.Subst
module U = FStar.Syntax.Util
module BU = FStar.Compiler.Util

let main argv =
    BU.print_string "Initializing ...\n";
    try
        FStar.Main.setup_hooks();
        Pars.init() |> ignore;
        Norm.run_all ();
        if Unif.run_all () then () else exit 1;
        exit 0
    with Error(err, msg, r, _ctx) when not <| FStar.Options.trace_error() ->
         if r = FStar.Compiler.Range.dummyRange
         then BU.print_string msg
         else BU.print2 "%s: %s\n" (FStar.Compiler.Range.string_of_range r) msg;
         exit 1
